src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts

```diff
@@ -404,6 +422,42 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit {
       data => {
         SELF.updateLeaderboardResults(data['results'], SELF);
         SELF.startLeaderboard(SELF.selectedPhaseSplit['id']);
+        SELF.sortLeaderboardTextOption = SELF.showLeaderboardByLatest ? 'Sort by best' : 'Sort by latest';
+      },
+      err => {
+        SELF.globalService.handleApiError(err);
+      },
+      () => {}
+    );
+  }
+
+  showLeaderboardByLatestOrBest() {
+    const API_PATH = this.endpointsService.particularChallengePhaseSplitUrl(this.selectedPhaseSplit['id']);
+    const SELF = this;
+    let BODY;
+
+    SELF.showLeaderboardByLatest = !SELF.showLeaderboardByLatest;
+
+    if (this.phaseSplits[this.selectedPhaseSplit['id']].hasOwnProperty('show_leaderboard_by_latest_submission')) {
+      BODY = JSON.stringify({
+        'show_leaderboard_by_latest_submission': this.phaseSplits[this.selectedPhaseSplit['id']].show_leaderboard_by_latest_submission
+      });
+    } else {
+      BODY = JSON.stringify({
+        'show_leaderboard_by_latest_submission': SELF.showLeaderboardByLatest
+      });
+    }
+
+    SELF.leaderboard = [];
+    SELF.showLeaderboardUpdate = false;
```

Comment:
Also we don't need this here as it is already gets set on the `refreshLeaderboard` function.

Reply:
Changed!